HeterogeneousCore/AlpakaInterface/test/alpaka/testPrefixScan.dev.cc

```diff
@@ -0,0 +1,212 @@
+    alpaka::syncBlockThreads(acc);
+
+    for_each_element_in_block_strided(acc, size, 1u, [&](uint32_t i) {
+      // TODO: handle both cases (int/float)
+      if constexpr (!std::is_floating_point_v<T>)
+        if (!((c[i] == c[i - 1] + 1) && (c[i] == i + 1) && (c[i] == co[i])))
+          printf("c[%d]=%d, co[%d]=%d\n", i, c[i], i, co[i]);
+      ALPAKA_ASSERT_OFFLOAD(c[i] == c[i - 1] + 1);
+      ALPAKA_ASSERT_OFFLOAD(c[i] == i + 1);
+      ALPAKA_ASSERT_OFFLOAD(c[i] == co[i]);
+    });
+  }
+};
+
+/*
+ * NB: GPU-only, so do not care about elements here.
+ */
+template <typename T>
+struct testWarpPrefixScan {
+  template <typename TAcc>
+  ALPAKA_FN_ACC void operator()(const TAcc& acc, uint32_t size) const {
+    if constexpr (!requires_single_thread_per_block_v<TAcc>) {
+      ALPAKA_ASSERT_OFFLOAD(size <= 32);
+      auto& c = alpaka::declareSharedVar<T[1024], __COUNTER__>(acc);
+      auto& co = alpaka::declareSharedVar<T[1024], __COUNTER__>(acc);
+
+      uint32_t const blockDimension = alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0u];
+      uint32_t const blockThreadIdx = alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u];
+      auto i = blockThreadIdx;
+      c[i] = 1;
+      alpaka::syncBlockThreads(acc);
+      auto laneId = blockThreadIdx & 0x1f;
+
+      warpPrefixScan(acc, laneId, c, co, i);
+      warpPrefixScan(acc, laneId, c, i);
+
+      alpaka::syncBlockThreads(acc);
+
+      ALPAKA_ASSERT_OFFLOAD(1 == c[0]);
+      ALPAKA_ASSERT_OFFLOAD(1 == co[0]);
+      if (i != 0) {
+        if (c[i] != c[i - 1] + 1)
+          printf(format_traits<T>::failed_msg, size, i, blockDimension, c[i], c[i - 1]);
+        ALPAKA_ASSERT_OFFLOAD(c[i] == c[i - 1] + 1);
+        ALPAKA_ASSERT_OFFLOAD(c[i] == static_cast<T>(i + 1));
+        ALPAKA_ASSERT_OFFLOAD(c[i] == co[i]);
+      }
+    } else {
+      // We should never be called outsie of the GPU.
+      ALPAKA_ASSERT_OFFLOAD(false);
+    }
+  }
+};
+
+struct init {
+  template <typename TAcc>
+  ALPAKA_FN_ACC void operator()(const TAcc& acc, uint32_t* v, uint32_t val, uint32_t n) const {
+    for_each_element_in_grid_strided(acc, n, [&](uint32_t index) {
+      v[index] = val;
+
+      if (index == 0)
+        printf("init\n");
+    });
+  }
+};
+
+struct verify {
+  template <typename TAcc>
+  ALPAKA_FN_ACC void operator()(const TAcc& acc, uint32_t const* v, uint32_t n) const {
+    for_each_element_in_grid_strided(acc, n, [&](uint32_t index) {
+      ALPAKA_ASSERT_OFFLOAD(v[index] == index + 1);
+
+      if (index == 0)
+        printf("verify\n");
+    });
+  }
+};
+
+int main() {
+  // get the list of devices on the current platform
+  auto const& devices = cms::alpakatools::devices<Platform>();
+  // auto const& host = cms::alpakatools::host();
+
+  if (devices.empty()) {
+    std::cout << "No devices available on the platform " << EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE)
+              << ", the test will be skipped.\n";
+    return 0;
+  }
+
```

Comment:
Should this be removed, and scram be made to run the test only if a device is available ?